requires.py

```diff
@@ -28,19 +28,22 @@ def changed(self):
         # assume we have all the data if we have the port
         if conv.get_remote('port'):
             conv.set_state('{relation_name}.available')
-        
+
     @hook('{requires:elasticsearch}-relation-{broken, departed}')
     def broken(self):
         self.remove_state('{relation_name}.connected')
         self.remove_state('{relation_name}.connected')
         self.set_state('{relation_name}.broken')
 
+    def list_unit_data(self):
+        ''' Iterate through all ElasticSearch conversations and return the data
+        for each cached conversation. This allows us to build a cluster string
+        directly from the relation data. eg:
 
-
-    def cluster_name(self):
-        self.get_remote('cluster-name')
-
-    def list_units(self):
+        for unit in elasticsearch.list_data():
+            print(unit['cluster_name'])
+        '''
         for conv in self.conversations():
-            unit = conv.scope
-            yield unit
+            yield {'cluster_name': conv.get_remote('cluster_name'),
```

Comment:
Correct.  `auto_accessors` is only valid for `GLOBAL` scope; it is equivalent to `self.conversation().get_remote(attr)` and depends on a default conversation, which only works if you happen to be in a hook context (not something that you should ever rely on unless you're in a `@hook` handler) or only have exactly 1 conversation (i.e., `GLOBAL` scope).  Otherwise, calling an `auto_accessor` method (or `self.get_remote()`) will fail with an error.

For `SERVICE` or `UNIT` scopes, you must explicitly loop over the conversations which will represent either all units of a service (`SERVICE` scope), or a single unit (`UNIT` scope).  For `UNIT` scope, `conv.get_remote(key)` is equivalent to `relation_get(key)` with the correct value for the `unit` and `rel_id` params.  For `SERVICE` scope, it will pick the value set by any unit of that service which has set a value, so they should either all (eventually) agree or the leader should set the value.